atef/procedure.py

```diff
@@ -216,27 +242,41 @@ class CodeStep(ProcedureStep):
 @dataclass
 class PlanOptions:
     """Options for a bluesky plan scan."""
-    #: The plan name.
+    #: Name to identify this plan
+    name: str
+    #: The plan name.  Bluesky plan or otherwise
     plan: str
     #: Plan arguments dictionary - argument name to value.
-    args: Sequence[Any]
+    args: Optional[Sequence[Any]] = field(default_factory=list)
     #: Plan keyword  arguments dictionary - argument name to value.
-    kwargs: Dict[Any, Any]
+    kwargs: Optional[Dict[Any, Any]] = field(default_factory=dict)
```

Comment:
Opinion: I find that if you specify `Optional` in the dataclass, you may as well initialize it to `None` instead of default-factory-generated empty dict

Reply:
I guess I wanted it to be optional from a user's point of view (you don't have to provide kwargs at initialization), but it's not optional from the dataclass' point of view (`None` can't be passed as kwargs).  

The right thing to do here is just provide a default, and not mark this as optional